File: scribbleshare-room/src/main/java/net/stzups/scribbleshare/room/server/websocket/protocol/server/messages/ServerMessageGetInvite.java
package net.stzups.scribbleshare.room.server.websocket.protocol.server.messages;

import io.netty.buffer.ByteBuf;
import net.stzups.scribbleshare.data.objects.InviteCode;
import net.stzups.scribbleshare.room.server.websocket.protocol.server.ServerMessage;
import net.stzups.scribbleshare.room.server.websocket.protocol.server.ServerMessageType;

public class ServerMessageGetInvite extends ServerMessage {
    private final String code;

    public ServerMessageGetInvite(InviteCode inviteCode) {
        code = inviteCode.getCode();
    }

    @Override
    protected ServerMessageType getMessageType() {
        return ServerMessageType.GET_INVITE;
    }

    @Override
    public void serialize(ByteBuf byteBuf) {
        super.serialize(byteBuf);
        writeString(code, byteBuf);
    }
}

File: scribbleshare-app/scribbleshare-frontend/src/scripts/protocol/server/messages/ServerMessageGetInvite.ts
import ServerMessage from "../ServerMessage.js";
import ServerMessageType from "../ServerMessageType.js";
import ByteBuffer from "../../ByteBuffer.js";

export default class ServerMessageGetInvite extends ServerMessage {
    code: string;

    constructor(byteBuffer: ByteBuffer) {
        super();
        this.code = byteBuffer.readString8();
    }

    getType(): ServerMessageType {
        return ServerMessageType.GET_INVITE;
    }
}